python/beeai_framework/adapters/ollama/backend/chat.py

```diff
@@ -30,8 +30,13 @@ def provider_id(self) -> ProviderName:
     def __init__(self, model_id: str | None = None, settings: dict | None = None) -> None:
         _settings = settings.copy() if settings is not None else None
 
-        if _settings is not None and not hasattr(_settings, "base_url") and "OLLAMA_BASE_URL" in os.environ:
-            _settings["base_url"] = os.getenv("OLLAMA_BASE_URL")
+        ollama_base_url_from_env = os.getenv("OLLAMA_BASE_URL")
+
+        if ollama_base_url_from_env:
+            if _settings is None:
+                _settings = {"base_url": ollama_base_url_from_env}
+            elif _settings is not None and "base_url" not in _settings:
+                _settings["base_url"] = ollama_base_url_from_env
```

Comment:
since the other providers should be able to do this. should this be generalized and moved into the `LiteLLMChatModel.__init__` method? for example,

```python
        base_url_from_env = os.getenv(f"{self.provider_id.upper()}_BASE_URL")

        if base_url_from_env:
            if _settings is None:
                _settings = {"base_url": base_url_from_env}
            elif _settings is not None and "base_url" not in _settings:
                _settings["base_url"] = base_url_from_env
```

Reply:
I'm not sure this will work?  Each adapter abstracts the connection details of the backend service so for Ollama we have BASE_URL but in Watsonx it's called WATSONX_URL and there's an additional project ID and API key that are required.  Each of these needs to be figured out on a per-adapter basis and the appropriate configuration added to the settings dictionary that's passed to the LiteLLMChatModel superclass.

Aside: Ollama BASE_URL may be better renamed API_BASE since that's how LiteLLM seem to call Ollama in their SDK.

Reply:
I may have mis-understood LiteLLM.  Leave it with me and I'll either merge this later or figure out a way to do it generically as per your suggestion.